src/components/Header.jsx

```diff
@@ -5,19 +5,43 @@ import { useEffect, useState } from 'react';
 import { useMediaQuery } from 'react-responsive';
 
 function Header() {
+  const [isHeaderVisible, setIsHeaderVisible] = useState(true);
+  const location = useLocation();
+  const isMobile = useMediaQuery({ maxWidth: 767 });
+  const isPostPage = location.pathname.startsWith('/post');
+
+  // 모바일이면서 포스트 페이지일 때 헤더가 나타나지 않도록 설정
+  useEffect(() => {
+    const handleHeaderVisibility = () => {
+      const isHeaderHidden = isMobile && isPostPage;
+      setIsHeaderVisible(!isHeaderHidden);
+    };
+
+    handleHeaderVisibility();
+
+    window.addEventListener('resize', handleHeaderVisibility);
+
+    return () => {
+      window.removeEventListener('resize', handleHeaderVisibility);
+    };
+  }, [isMobile, isPostPage]);
+
   return (
-    <header className={styles.header}>
-      <div className={styles.container}>
-        <Link to='/' className={styles.wrapper}>
-          <img src={logo} className={styles.logo} alt='로고' />
-          <h1 className={styles.logo}>Rolling</h1>
-        </Link>
-        <Link to='/post' className={styles.button}>
-          롤링 페이퍼 만들기
-        </Link>
-      </div>
-    </header>
+    isHeaderVisible && (
+      <header className={styles.header}>
+        <div className={styles.container}>
+          <Link to='/' className={styles.wrapper}>
+            <img src={logo} className={styles.logo} alt='로고' />
+            <h1 className={styles.logo}>Rolling</h1>
+          </Link>
+          {/* 포스트 페이지가 아닐 때 롤링 페이퍼 만들기 버튼이 나타나지 않도록 설정 */}
+          {!isPostPage && (
+            <Link to='/post' className={styles.button}>
+              롤링 페이퍼 만들기
+            </Link>
```

Comment:
Link로 스타일 적용하는거 좋지만..! 
버튼은 한주님이 만들어두신 걸로 변경하는게 좋을거 같아요 ㅎㅎ
```suggestion
<Button type='outlined' size='40'>롤링 페이퍼 만들기</Button>
```

고정 height가 적용되어 있어서 42px도 따로 지정해주면 될거같아요
Link는 버튼에 넣거나..? 아니면 onClick 으로 핸들러 함수 지정..!